### ISSUE TITLE
Please add support for dynamic JPA 2.1 Entity Graph

### ISSUE THREAD

**Issue by throwable** [my2iu/Jinq#73]
Something like this:

`myEntityStream.fetch(me -> Arrays.asList(me.getProperty1(), me.getProperty2(), me.getEntity().getProperty()))`

That will construct an Entity Graph and apply to the query:

```
EntityGraph<MyEntity> eg = em.createEntityGraph(MyEntity.class);
eg.addAttributeNodes("property1");
eg.addAttributeNodes("property2");
eg.addAttributeNodes("entity.property");
...
Properties props = new Properties();
props.put("javax.persistence.fetchgraph", eg);
...
```

**Comment by my2iu** [comment]
I'm no longer doing any active development on Jinq, so I don't have the cycles to look at this, but I'll keep the issue around in case the situation changes.

**Comment by ronelm2000** [comment]
Another slightly less intrusive fix for this should be a way to propagate JINQ Hints to JPA.

```java
...
EntityGraph<MyEntity> eg = em.createEntityGraph(MyEntity.class);
eg.addAttributeNodes("property1");
eg.addAttributeNodes("property2");
eg.addAttributeNodes("entity.property");

myEntityStream = myEntityStream.setHint("javax.persistence.fetchgraph", eg);
...
```
As far as I'm know FetchGraphs in JPA 2.1 onwards are just treated as hints and fetch has a clearly defined output (using fetch joins). At most this has to be a new API and not on existing .fetchJoin methods. `stream.fetchGraph(me -> {me.getProperty1(); me.getProperty2()

**Comment by my2iu** [comment]
Ok, implemented. I'm not entirely sure it's working as intended though. EclipseLink didn't seem to do anything with a small query that I tried to use it with, and Hibernate did do something, but the result seemed really sub-optimal.

**Comment by my2iu** [comment]
This is now in the 1.8.26 release of Jinq.

**Comment by ronelm2000** [comment]
@my2iu found the problem:

```java
      else if ("javax.persistence.fetchgraph".equals("name") && val instanceof EntityGraph)
         javaxPersistenceFetchgraph = (EntityGraph<?>)val;
```

**Comment by my2iu** [comment]
You're right. Sorry about that. I was rushing through things and never bothered tracking down why the test case wasn't causing EclipseLink to generate different SQL than before. That change, plus some strange incantations to get EclipseLink weaving working under Maven, seems to cause EclipseLink to generate different SQL.

The fix has now been released in 1.8.27.
